File: src/CrimeWatch.Domain/AggregateModels/WitnessAggregate/Witness.cs
using CrimeWatch.Domain.AggregateModels.AccountAggregate;
using CrimeWatch.Domain.AggregateModels.UserAggregate;

namespace CrimeWatch.Domain.AggregateModels.WitnessAggregate;
public class Witness : AggregateRoot<WitnessId>
{
    public UserId UserId { get; set; } = default!;
    public AccountId AccountId { get; set; } = default!;

    public Account? Account { get; set; }
    public User? User { get; set; }

    public static Witness Create(
        string firstName,
        string lastName,
        Gender gender,
        DateOnly dateOfBirth,
        string phoneNumber,
        string email,
        string password
        )
    {
        User user = User.Create(firstName, lastName, gender, dateOfBirth, phoneNumber);
        Account account = Account.Create(email, password, false);

        return new()
        {
            Id = new(Guid.NewGuid()),
            User = user,
            Account = account
        };
    }

    public Witness Update(
        string firstName,
        string lastName,
        Gender gender,
        DateOnly dateOfBirth,
        string email,
        string password,
        string phoneNumber)
    {
        User!.FirstName = firstName;
        User!.LastName = lastName;
        User!.DateOfBirth = dateOfBirth;
        User!.PhoneNumber = phoneNumber;
        User!.Gender = gender;
        Account!.Email = email;
        Account!.Password = password;

        return this;
    }
}

File: src/crimewatch.web.client/src/models/Witness.ts
import { Gender } from "../enums/Gender";
import AccountId from "../valueObjects/AccountId";
import UserId from "../valueObjects/UserId";
import WitnessId from "../valueObjects/WitnessId";
import Account from "./Account";
import User from "./User";

export interface UpdateWitnessDto {
    id: WitnessId;
    firstName: string;
    lastName: string;
    gender: Gender;
    dateOfBirth: Date;
    phoneNumber: string;
    email: string;
    Password: string;
}

export interface CreateWitnessDto {
    firstName: string;
    lastName: string;
    gender: Gender;
    birthDate: Date;
    phoneNumber: string;
    email: string;
    password: string;
}

export default interface Witness {
    witnessId: WitnessId;
    userId: UserId;
    accountId: AccountId;
    account: Account | null;
    user: User | null;
}